### PR TITLE
Fix/audit report

### PR DESCRIPTION
**Pull request by hoonsubin**
added input validation for `btcUnlockTx()`, added more comments, added additional tests, unified bitcoin network enum to use it from `bitcoinjs-lib` rather than having a custom enum.

### CODE REVIEW

File: src/helpers/lockdrop/BitcoinLockdrop.ts [hunk 1]
```diff
@@ -222,27 +274,35 @@ export function btcLockScript(publicKeyHex: string, blockSequence: number): Buff
...
 export function btcUnlockTx(
     signer: Signer,
     network: Network,
     lockTx: UnspentTx,
     lockScript: Buffer,
-    lockBlocks: number,
-    recipient: string,
+    blockSequence: number,
+    recipientAddress: string,
     fee: number, // satoshis
```
Comment (satellitex) [inline src/helpers/lockdrop/BitcoinLockdrop.ts:306]: In the function btcUnlockTx(), the parameters lockBlocks and fee should be ensured to be non-negative.

Comment (hoonsubin) [reply src/helpers/lockdrop/BitcoinLockdrop.ts:306]: fixed in https://github.com/staketechnologies/lockdrop-ui/pull/63/commits/2cbb55127419765528ee9d6c771812f5d63ca12d, please check

**Comment by hoonsubin** [comment]
> Audit says:
> 2. For “BTC lock script tests”, add tests that check the maximum and minimum intended durations.
> 
> Could you add the invalid and valid min/max cases:
> duration = `29<=`(invalid) or `30`(valid) or `300`(valid) or `301>=`(invalid).

in terms of client interaction, the dropdown menu does limit the days the user can choose from in here https://github.com/staketechnologies/lockdrop-ui/blob/fix/audit-report/src/data/lockInfo.ts#L36

Should I explicitly add the limits in this function too? https://github.com/staketechnologies/lockdrop-ui/blob/fix/audit-report/src/helpers/lockdrop/BitcoinLockdrop.ts#L280

**Comment by satellitex** [comment]
@hoonsubin 
> in terms of client interaction, the dropdown menu does limit the days the user can choose from in here 
> https://github.com/staketechnologies/lockdrop-ui/blob/fix/audit-report/src/data/lockInfo.ts#L36
Ok! think no problem that.

> Should I explicitly add the limits in this function too? https://github.com/staketechnologies/lockdrop-ui/blob/fix/audit-report/src/helpers/lockdrop/BitcoinLockdrop.ts#L280

Maybe, `-1` test is needed.

**Comment by hoonsubin** [comment]
@satellitex looks like I added way too many day/block sequence checks. So I refactored and made sure that only functions that create the lock script will validate their inputs while helper functions like `daysToBlockSequence()` will not check the duration. This should make things easier to understand and secure enough

**Comment by hoonsubin** [comment]
@akru @satellitex 
Here's the list of all the audit fixes made on the lockdrop UI side, please confirm them if they are correct.

-----

## QSP-2 Min and max limits not known and not enforced at the domain level

fix: `BitcoinLockdrop.ts`

Also note that lack of enforcement of max with a low probability could cause an overflow here

Fixes:

- <https://github.com/staketechnologies/lockdrop-ui/pull/63/files#diff-7c6bfd88869aa4e403f750d4ce60b5d0R278>
- <https://github.com/staketechnologies/lockdrop-ui/blob/fix/audit-report/src/helpers/lockdrop/BitcoinLockdrop.ts#L246>

## QSP-3 Integer division not enforced

fix: `BitcoinLockdrop.ts`

It is still a good practice to make sure the input to the BitcoinLockdrop.ts L120 is finite or even integer.

Fixes:

- <https://github.com/staketechnologies/lockdrop-ui/blob/fix/audit-report/src/helpers/lockdrop/BitcoinLockdrop.ts#L243>
- <https://github.com/staketechnologies/lockdrop-ui/blob/fix/audit-report/src/helpers/lockdrop/BitcoinLockdrop.ts#L323>
- <https://github.com/staketechnologies/lockdrop-ui/pull/63/files#diff-7c6bfd88869aa4e403f750d4ce60b5d0R218>

## QSP-4 Input parameters not validated

fix: `BitcoinLockdrop.ts`

If publicKeyHex is incorrect or invalid, the BTC could remain locked forever.<br />
Fix: <https://github.com/staketechnologies/lockdrop-ui/pull/63/files#diff-7c6bfd88869aa4e403f750d4ce60b5d0R251>

In the function btcUnlockTx(), the parameters lockBlocks and fee should be ensured to be non-negative.<br />
Fix: <https://github.com/staketechnologies/lockdrop-ui/pull/63/files#diff-7c6bfd88869aa4e403f750d4ce60b5d0R319>

In the function getLockP2SH(), a check should be added ensuring that the duration is non negative.<br />
Fix: <https://github.com/staketechnologies/lockdrop-ui/pull/63/files#diff-7c6bfd88869aa4e403f750d4ce60b5d0R278>

## QSP-5 Test coverage improvements

fix: `BitcoinLockdrop.ts`

1. Adding handling the corner cases for “BTC signature tests”: testing cases when signature is invalid.<br />
Fix: <https://github.com/staketechnologies/lockdrop-ui/blob/fix/audit-report/src/tests/btcLock.test.ts#L92>

2. For “BTC lock script tests”, add tests that check the maximum and minimum intended durations.<br />
Fix: <https://github.com/staketechnologies/lockdrop-ui/blob/fix/audit-report/src/tests/btcLock.test.ts#L195>

3. For “BTC lock script tests”, introduce an adversary Bob who attempts to unlock Alice’s bitcoins and confirm that it is not possible.<br />
Fix:<https://github.com/staketechnologies/lockdrop-ui/blob/fix/audit-report/src/tests/btcLock.test.ts#L295>

4. For “BTC lock script tests”, add tests for invalid / unset values for and pubkey.<br />
Fix: <https://github.com/staketechnologies/lockdrop-ui/blob/fix/audit-report/src/tests/btcLock.test.ts#L214>

5. Several functions in BitcoinLockdrop.ts appear to not be covered by tests.<br />
Fix: <https://github.com/staketechnologies/lockdrop-ui/blob/fix/audit-report/src/tests/btcLock.test.ts#L132>
